src/extensions/additional/GPT/MarkupGpt/plugin.ts

```diff
@@ -0,0 +1,168 @@
+
+                            if (from === to) {
+                                this.disablePromptPresets = true;
+
+                                if (isEmptyGptPrompts(gptProps, true)) return;
+
+                                const decorationWidget = Decoration.widget({
+                                    widget: new SpanWidget(WIDGET_DECO_CLASS_NAME, ' '),
+                                });
+
+                                this.decos = Decoration.set([decorationWidget.range(from)]);
+
+                                return;
+                            }
+
+                            this.disablePromptPresets = false;
+
+                            if (isEmptyGptPrompts(gptProps, false)) return;
+
+                            this.decos = Decoration.set([
+                                {
+                                    from,
+                                    to,
+                                    value: Decoration.mark({class: WIDGET_DECO_CLASS_NAME}),
+                                },
+                            ]);
+                        }
+
+                        if (eff.is(HideMarkupGptEffect)) {
+                            this.decos = Decoration.none;
+                        }
+                    }
+                }
+            }
+
+            docViewUpdate() {
+                this._anchor = this._view.dom
+                    .getElementsByClassName(WIDGET_DECO_CLASS_NAME)
+                    .item(0);
+                this._renderItem.rerender();
+            }
+
+            destroy() {
+                this._clearSelectedText();
+                this._renderItem.remove();
+            }
+
+            renderPopup() {
+                if (!this._anchor || this.markup === null) {
+                    return null;
+                }
+
+                return renderPopup(this._anchor as HTMLElement, {
+                    ...gptProps,
+                    disablePromptPresets: this.disablePromptPresets,
+                    onClose: () => hideMarkupGpt(this._view),
+                    markup: this.markup,
+                    onApplyResult: (changedMarkup) => this._onApplyResult(changedMarkup),
+                });
+            }
+
+            _getDecorationText(update: ViewUpdate, from: number, to: number): string {
+                return update.state.doc.sliceString(from, to);
+            }
+
+            _clearSelectedText() {
+                this.markup = null;
+            }
+
+            _setSelectedText(str: string) {
+                this.markup = str;
+            }
+
+            _onApplyResult(changedMarkup: string) {
+                const {from, to} = this.selectedPosition;
+                const changes = [{from: from, to: to, insert: changedMarkup}];
+
+                const transaction = this._view.state.update({
+                    changes: changes,
+                });
+
+                this._view.dispatch(transaction);
+                hideMarkupGpt(this._view);
```

Comment:
You can add `{effects: [HideMarkupGptEffect.of(null)]}` to previous transaction and not call dispatch twice.